### PR TITLE
Add dynamic import to require() transform

### PR DESCRIPTION
**Pull request by justinfagnani**
Based on #347. Will rebase when that's merged.

 - [ ] CHANGELOG.md has been updated

### CODE REVIEW

File: src/babel-plugin-dynamic-import-amd.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ */
+
+import dyanamicImportSyntax from '@babel/plugin-syntax-dynamic-import';
+import {NodePath} from '@babel/traverse';
+import {CallExpression, Identifier, Program, Statement} from 'babel-types';
+import template from '@babel/template';
+
+const ast = template.ast;
+
+export const dynamicImportAMD = {
```
Comment (aomarks) [inline src/babel-plugin-dynamic-import-amd.ts]: add a description

Comment (aomarks) [inline src/babel-plugin-dynamic-import-amd.ts]: s/AMD/Amd?

/ducks

File: src/babel-plugin-dynamic-import-amd.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      }
+      let requireId: Identifier|undefined = undefined;
+      do {
+        requireId = path.scope.generateUidIdentifier('require');
+      } while (identifiers.has(requireId.name));
+      const statements = path.node.body as Statement[];
+      statements.unshift(ast`import * as ${requireId} from 'require';`);
+      for (const importPath of dynamicImports) {
+        const specifier = importPath.node.arguments[0];
+        // The 2-argument local require takes an array of module IDs and
```
Comment (aomarks) [inline src/babel-plugin-dynamic-import-amd.ts]: This comment seems more relevant just above, before you add the require import.

File: src/babel-plugin-dynamic-import-amd.ts (hunk 1 above)
Comment (justinfagnani) [reply src/babel-plugin-dynamic-import-amd.ts]: done

File: src/babel-plugin-dynamic-import-amd.ts (hunk 2 above)
Comment (justinfagnani) [reply src/babel-plugin-dynamic-import-amd.ts]: updated comments

File: src/babel-plugin-dynamic-import-amd.ts (hunk 1 above)
Comment (justinfagnani) [reply src/babel-plugin-dynamic-import-amd.ts]: done

**Pull request merged**